### DISCUSSION TITLE
Node3DEditorViewport should be exposed to GDScript and it should be possible to set the camera property

### DISCUSSION THREAD

**Discussion by blackears** [godotengine/godot-proposals#9983] [Editor]
I'm writing an addon and would like to have the ability to set the viewport camera position explicitly.  As far as I know this is not currently possible.  I think it would be a good idea to make Node3DEditorViewport accessible through the API and also to let users set it's camera property so that addons can control the camera.

**Comment by AThousandShips** [comment]
You can get the 3D viewport with [`get_editor_viewport_3d`](https://docs.godotengine.org/en/latest/classes/class_editorinterface.html#class-editorinterface-method-get-editor-viewport-3d) and access its camera

**Comment by blackears** [comment]
That doesn't help.  If I set the transform of the camera, it is immediately overwritten when I next move the camera.

In the attached project, you can install the addon and then press F3 when the 3D editor viewport is displayed.  While the viewport camera is updated for the next frame, as soon as you try to tumble the viewport or right-click fly, whatever you set it to is forgotten and the camera snaps back to the position it was before.

[hack-editor-viewport-camera.zip](https://github.com/user-attachments/files/15888209/hack-editor-viewport-camera.zip)

**Comment by blackears** [comment]
I created an issue for this.

https://github.com/godotengine/godot/issues/93341
